pinot-core/src/test/java/org/apache/pinot/queries/RangeQueriesTest.java

```diff
@@ -232,4 +268,38 @@ public void testCountOverRangeFilter(String query, int expectedCount) {
     assertEquals(aggregationResult.size(), 1);
     assertEquals(((Number) aggregationResult.get(0)).intValue(), expectedCount, query);
   }
+
+  @Test(dataProvider = "countTestCases")
+  public void testCountOverRangeFilterAfterReload(String query, int expectedCount)
```

Comment:
- SELECT <particularColumn> FROM foo LIMIT <everything> -- this will test out that dict is correct and fwd index got rewritten
- Try the same in GROUP BY as GROUP BY uses dict. 
- Try MIN and MAX aggregations -- they can be answered from dict. 
- WHERE clause like mentioned in another commend